projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts

```diff
@@ -0,0 +1,287 @@
+import { Component, DebugElement, ViewChild } from '@angular/core';
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+import { FormControl, FormGroup, ReactiveFormsModule } from '@angular/forms';
+import { By } from '@angular/platform-browser';
+import { of } from 'rxjs';
+import { WriteAddressFieldComponent } from '..';
+import { AddressModel } from '../../models';
+import { AddressService } from '../../services';
+
+describe('WriteAddressFieldComponent', () => {
+
+  const POSTCODE = 'P05T CDE';
+
+  const $POSTCODE_LOOKUP = By.css('.postcodeLookup');
+  const $POSTCODE_LOOKUP_INPUT = By.css('.postcodeinput');
+  const $POSTCODE_LOOKUP_FIND = By.css('.postcodeLookup > button');
+  const $POSTCODE_LOOKUP_ERROR_MESSAGE = By.css('.error-message');
+
+  const $SELECT_ADDRESS = By.css('#selectAddress');
+  const $ADDRESS_LIST = By.css('#selectAddress > .addressList');
+
+  const $MANUAL_LINK = By.css('.manual-link');
+
+  @Component({
+    selector: `write-address-field`,
+    template: `<xuilib-write-address-field [formGroup]="formGroup">
+    </xuilib-write-address-field>`
+  })
+  class WrapperComponent {
+    @ViewChild(WriteAddressFieldComponent, /* TODO: add static flag */ {})
+    public componentUnderTest: WriteAddressFieldComponent;
+    public addressField = new FormControl({});
+    public formGroup = addressFormGroup();
+  }
+
+  let addressService: AddressService;
+  let wrapperComponent: WrapperComponent;
+  let debugElement: DebugElement;
+  let fixture: ComponentFixture<WrapperComponent>;
+
+  function addressFormGroup() {
+    return new FormGroup({
+      addressLine1: new FormControl(),
+      addressLine2: new FormControl(),
+      addressLine3: new FormControl(),
+      postTown: new FormControl(),
+      county: new FormControl(),
+      postCode: new FormControl(),
+      country: new FormControl()
+    });
+  }
+
+  function buildAddress(entryNo: number): AddressModel {
+    const address = new AddressModel();
+    address.addressLine1 = `addressLine1-${entryNo}`;
+    address.addressLine2 = `addressLine2-${entryNo}`;
+    address.addressLine3 = `addressLine3-${entryNo}`;
+    address.postTown = `postTown-${entryNo}`;
+    address.county = `county-${entryNo}`;
+    address.postCode = `postCode-${entryNo}`;
+    address.country = `country-${entryNo}`;
+    return address;
+  }
+
+  function queryPostcode(postcode: string) {
```

Comment:
@johnbenjamin-hmcts - could do with a return type

Reply:
@Daniel-Lam - I have updated the code.